engine/class_modules/sc_mage.cpp

```diff
@@ -3551,15 +3550,20 @@ struct arcane_barrage_t final : public dematerialize_spell_t
     am *= 1.0 + p()->buffs.aethervision->check_stack_value();
     am *= 1.0 + p()->buffs.arcane_harmony->check_stack_value();
     am *= 1.0 + p()->buffs.intuition->check_value();
+    am *= 1.0 + p()->buffs.nether_precision->check_value();
+    am *= 1.0 + p()->buffs.arcane_soul_counter->check_stack_value();
 
     return am;
   }
 
   void impact( action_state_t* s ) override
   {
     dematerialize_spell_t::impact( s );
-    if ( result_is_hit( s->result ) )
+    if ( result_is_hit( s->result ) ) 
       trigger_glorious_incandescence( s->target );
+
+    if ( ( s->n_targets > 2 ) && result_is_hit( s->result ) && p()->talents.arcane_rebound.ok() )
```

Comment:
This would trigger the Rebound on *every* target rather than just the primary one.

Reply:
Also, hardcoding the target threshold is bad practice.